packages/query-core/src/hydration.ts

```diff
@@ -142,10 +143,17 @@ export function hydrate(
   queries.forEach((dehydratedQuery) => {
     const query = queryCache.get(dehydratedQuery.queryHash)
 
+    // Reset fetch status to idle in the dehydrated state to avoid
+    // query being stuck in fetching state upon hydration
+    const dehydratedQueryState = {
+      ...dehydratedQuery.state,
+      fetchStatus: 'idle' as FetchStatus,
```

Comment:
```suggestion
      fetchStatus: 'idle' as const,
```

Reply:
the problem here is that `'foo' as FetchStatus` would also be fine for typescript, so if we ever rename `idle` to something else, TS will not notify us about this change here. `const assertions` solve this: https://tkdodo.eu/blog/the-power-of-const-assertions

Reply:
Yes makes sense, updated.